src/wasm/wasm-emscripten.cpp

```diff
@@ -207,16 +207,16 @@ struct JSCallWalker : public PostWalker<JSCallWalker> {
     }
     const auto& tableSegmentData = wasm.table.segments[0].data;
 
+    jsCallStartIndex =
+        wasm.table.segments[0].offset->cast<Const>()->value.getInteger();
     // Check if jsCalls have already been created
     for (Index i = 0; i < tableSegmentData.size(); ++i) {
       if (tableSegmentData[i].startsWith("jsCall_")) {
-        jsCallStartIndex = i;
+        jsCallStartIndex += i;
         return;
       }
     }
-    jsCallStartIndex =
-        wasm.table.segments[0].offset->cast<Const>()->value.getInteger() +
-        tableSegmentData.size();
+    jsCallStartIndex += tableSegmentData.size();
```

Comment:
So before this part was
```
jsCallStartIndex = wasm.table.segments[0].offset->cast<Const>()->value.getInteger() + tableSegmentData.size();
```
which is executed when you first call jsCalls, and this respects avoiding the 0 index.

Reply:
Nevermind, I was confused..